File: N8N/python-backend/Workers/nodes/runNode/runner.py
from typing import Any, Dict
from fastapi import HTTPException
from ..agent import run_agent
from ..email import send_Email
from ..form import run_form
from ..telegram import send_Telegram_Msg


async def runNode(node: Any, context: Dict[str, Any]):
    try:
        node_type = node.get("type")
        if node_type == "email":
            return await send_Email(node["credentialId"], node["template"], context)
        elif node_type == "telegram":
            return await send_Telegram_Msg(
                node["credentialId"], node["template"], context
            )
        elif node_type == "form":
            return await run_form(node["credentialId"], node["template"], context)
        elif node_type == "agent":
            return await run_agent(node["credentialId"], node["template"], context)
        else:
            raise HTTPException(
                status_code=400, detail=f"Node type is not found: {node_type}"
            )
    except HTTPException:
        raise
    except Exception:
        raise HTTPException(
            status_code=500, detail="Internal server error while running code"
        )

File: N8N/apps/workers/nodes/runNodes/runner.ts
import sendTelegramMessage from "../telegram";
import sendEmail from "../resendEmail";

export async function runNode(node: any, Context: Record<string, any>) {
  try {
    switch (node.type) {
      case "ResendEmail":
        return await sendEmail(node.template, node.credentialId, Context);
      case "Telegram":
        return await sendTelegramMessage(
          node.template,
          node.credentialId,
          Context,
        );
      default:
        throw new Error(`Unknown node type: ${node.type}`);
    }
  } catch (error) {
    console.log("Unkown error while running node:", error);
  }
}
